src/WebJobs.Script.Scaling/ScaleUtils.cs

```diff
@@ -109,24 +109,32 @@ public static string GetToken(DateTime expiredUtc)
                         cs.FlushFinalBlock();
                     }
 
-                    return string.Format("{0}.{1}", iv, Convert.ToBase64String(ms.ToArray()));
+                    return string.Format("{0}.{1}.{2}", iv, Convert.ToBase64String(ms.ToArray()), GetSHA256Base64String(aes.Key));
                 }
             }
         }
 
         public static void ValidateToken(string token)
         {
             var parts = token.Split(new[] { '.' }, StringSplitOptions.RemoveEmptyEntries);
-            if (parts.Length != 2)
+            if (parts.Length != 2 && parts.Length != 3)
             {
                 throw new ArgumentException("Malform encrypted data.");
             }
 
             var iv = Convert.FromBase64String(parts[0]);
             var data = Convert.FromBase64String(parts[1]);
+            var base64KeyHash = parts.Length == 3 ? parts[2] : null;
+
+            var encryptionKey = AppServiceSettings.RuntimeEncryptionKey;
+            if (!string.IsNullOrEmpty(base64KeyHash) && !string.Equals(GetSHA256Base64String(encryptionKey), base64KeyHash))
```

Comment:
should this condition also check for a `null` encryption key (as it will be what you'll get in environments where the app setting does not exist) and evaluate to `true` if that is that case so that will also throw the exception below?

Reply:
AppServiceSettings.RuntimeEncryptionKey already choked.